UI/PlayerListWnd.cpp

```diff
@@ -430,13 +430,20 @@ namespace {
                     }
                 }
 
-                for (auto* planet : objects.allRaw<Planet>()) {
-                    if (planet->Owner() == empire->EmpireID()) {
-                        empires_planet_count      += 1;
-                        empires_production_points += planet->GetMeter(MeterType::METER_INDUSTRY)->Initial();
-                        empires_research_points   += planet->GetMeter(MeterType::METER_RESEARCH)->Initial();
-                        empires_influence_points  += planet->GetMeter(MeterType::METER_INFLUENCE)->Initial();
-                    }
+                const auto owned_by_empire = [emp_id{empire->EmpireID()}](const UniverseObject& obj)
+                { return !obj.Unowned() && obj.OwnedBy(emp_id); };
+
+                empires_planet_count += objects.count<Planet>(owned_by_empire);
+
+                static constexpr auto initial_or_zero = [](const UniverseObject* obj, MeterType mt) {
+                    const auto* meter = obj->GetMeter(mt);
+                    return meter ? meter->Initial() : 0.0;
+                };
+
+                for (auto* obj : objects.findRaw(owned_by_empire)) {
```

Comment:
Based on what I saw, seems like those objects need to be checked against stale data (and likely against known destroyed, too). For my quick and dirty test I added that check into `owned_by_empire`  predicate

Reply:
specifically I did this and numbers matched:

`const auto owned_by_empire = [emp_id{ empire->EmpireID() }, this_client_known_destroyed_objects, this_client_stale_object_info](const UniverseObject& obj) {
                    return !obj.Unowned() && obj.OwnedBy(emp_id) && !this_client_known_destroyed_objects.contains(obj.ID())
                        && !this_client_stale_object_info.contains(obj.ID()); 
                    };`

Reply:
Probly make those lambda captures of the containers by reference...